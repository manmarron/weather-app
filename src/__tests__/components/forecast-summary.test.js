import React from 'react';
import { render } from "@testing-library/react";
import ForecastSummary from '../../components/forecast-summary';


describe("ForecastSummary", () => {

    it('renders', () => {
    const { asFragment } = render(
        <ForecastSummary
        Date="mockDate"
        temperature="mockTemperature"
        description="mockDescription"
        icon="mockIcon"
        />
      );
  
      expect(asFragment()).toMatchSnapshot();
    });

    it('renders the correct props', () => {
        const { getByText } = render(
            <ForecastSummary
            Date="mockDate"
            temperature="mockTemperature"
            description="mockDescription"
            icon="mockIcon"
            />
          );

      expect(getByText("mockDate")).toHaveClass("date");
      expect(getByText("mockTemperature")).toHaveClass("temperature");
      expect(getByText("mockDescription")).toHaveClass("description");
      expect(getByText("mockIcon")).toHaveClass("icon");
});

});
