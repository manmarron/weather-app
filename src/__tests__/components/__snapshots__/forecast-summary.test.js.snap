// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
  >
    , mockDescription, mockIcon, 
  </div>
</DocumentFragment>
`;
